openshift/dynamic/client.py

```diff
@@ -979,6 +979,16 @@ class ResourceInstance(object):
 
     def __init__(self, resource, instance):
         self.resource_type = resource
+        # If we have a list of resources, then set the apiVersion and kind of
+        # reach resource in 'items'
+        kind = instance['kind']
+        if kind.endswith('List') and 'items' in instance:
+            kind = instance['kind'][:-4]
+            for i in range(len(instance['items'])):
```

Comment:
Should we only do the update if the `apiVersion` and `kind` aren't set, just in case?

Reply:
I totally was intending to do that, do you think I should switch to just using field assignment and add a conditional for each?